pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -139,6 +139,9 @@ sys.exit('no disk found with OSD ID $OSD_ID')
 	# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag
 	ceph-volume raw activate --device "$DEVICE" --no-systemd --no-tmpfs
 fi
+
+# create new keyring
+ceph auth get-or-create osd.0 mon 'profile osd' mgr 'profile osd' osd 'allow *'
```

Comment:
```suggestion
ceph auth get-or-create osd.$OSD_ID mon 'profile osd' mgr 'profile osd' osd 'allow *'
```

I think this should be the first thing we do in the container.
Also, we must use the osd bootstrap key instead. Otherwise the current command won't work. You might need to pass monitors' addresses too.

Reply:
Ok it looks like we have the mon IP via https://github.com/rook/rook/blob/c929cb053674015e613100a8bfcd8af0f9032453/pkg/operator/ceph/cluster/osd/envs.go#L205.  So you might just need to call the Ceph CLI with a different user such as the osd-bootstrap, but we need to make sure we have that key mapped in the container too.

Reply:
@leseb for using different user `ceph auth get-or-create client.bootstrap-osd mon 'allow profile bootstrap-osd'` will work? and can you explain about mapping the key in the container?

Reply:
Will do but that depends on the error, when running this code, do you see any errors? If so please paste it.

Reply:
in the activate osd init-container 

```ceph auth get-or-create client.bootstrap-osd mon 'allow profile bootstrap-osd'
2021-05-31T11:40:01.706+0000 7f168f8c9700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-05-31T11:40:01.706+0000 7f168f8c9700 -1 AuthRegistry(0x7f168805ea70) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-05-31T11:40:01.707+0000 7f168f8c9700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-05-31T11:40:01.707+0000 7f168f8c9700 -1 AuthRegistry(0x7f168f8c7ea0) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-05-31T11:40:01.707+0000 7f168d665700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-05-31T11:40:01.707+0000 7f168f8c9700 -1 monclient: authenticate NOTE: no keyring found; disabled cephx authentication
[errno 13] RADOS permission denied (error connecting to the cluster)
```

Reply:
@leseb ^^

Reply:
You need to call like `ceph -n client.bootstrap-osd auth get-or-create osd.$OSD_ID mon 'profile osd' mgr 'profile osd' osd 'allow *'`.

If this is not working you might need to specify the path of the key with `-k /var/lib/ceph/bootstrap-osd/keyring` but you might need to mount that key as well.

Reply:
```+ ceph -n client.bootstrap-osd auth get-or-create osd.0 mon 'profile osd' mgr 'profile osd' osd 'allow *' -k /var/lib/ceph/bootstrap-osd/keyring
2021-06-10T16:04:39.626+0000 7f50e3fff700 -1 auth: error reading file: /var/lib/ceph/bootstrap-osd/keyring: bufferlist::read_file(/var/lib/ceph/bootstrap-osd/keyring): read error:(21) Is a directory
2021-06-10T16:04:39.626+0000 7f50e3fff700 -1 auth: failed to load /var/lib/ceph/bootstrap-osd/keyring: (21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: error reading file: /var/lib/ceph/bootstrap-osd/keyring: bufferlist::read_file(/var/lib/ceph/bootstrap-osd/keyring): read error:(21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: failed to load /var/lib/ceph/bootstrap-osd/keyring: (21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: error reading file: /var/lib/ceph/bootstrap-osd/keyring: bufferlist::read_file(/var/lib/ceph/bootstrap-osd/keyring): read error:(21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: failed to load /var/lib/ceph/bootstrap-osd/keyring: (21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: error reading file: /var/lib/ceph/bootstrap-osd/keyring: bufferlist::read_file(/var/lib/ceph/bootstrap-osd/keyring): read error:(21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 auth: failed to load /var/lib/ceph/bootstrap-osd/keyring: (21) Is a directory
2021-06-10T16:04:39.627+0000 7f50e9787700 -1 monclient: keyring not found
[errno 21] error connecting to the cluster
```

the error says `/var/lib/ceph/bootstrap-osd/keyring` is `Is a directory` but if I just mount `/var/lib/ceph/bootstrap-osd/` its says no file `keyring` found

Reply:
I think there is a bit more work to do at the end:

Option 1: put the bootstrap-osd key in a secret and mount it as VolumeSource 
Option 2: use the admin key which already available in a Secret

I prefer option 1 since option 2 is not secure. Option 1 is just a bit more work for you but a good exercise to discover more of the Rook's codebase too!